pkg/reconciler/pipelinerun/cancel.go

```diff
@@ -80,17 +114,21 @@ func cancelPipelineRun(ctx context.Context, logger *zap.SugaredLogger, pr *v1bet
 			continue
 		}
 	}
-	// If we successfully cancelled all the TaskRuns and Runs, we can consider the PipelineRun cancelled.
-	if len(errs) == 0 {
-		pr.Status.SetCondition(&apis.Condition{
-			Type:    apis.ConditionSucceeded,
-			Status:  corev1.ConditionFalse,
-			Reason:  ReasonCancelled,
-			Message: fmt.Sprintf("PipelineRun %q was cancelled", pr.Name),
-		})
-		// update pr completed time
-		pr.Status.CompletionTime = &metav1.Time{Time: time.Now()}
-	} else {
+
+	return errs
+}
+
+// gracefullyCancelPipelineRun marks any non-final resolved TaskRun(s) as cancelled and runs finally.
+func gracefullyCancelPipelineRun(ctx context.Context, logger *zap.SugaredLogger, pr *v1beta1.PipelineRun, pipelineRunFacts *resources.PipelineRunFacts, clientSet clientset.Interface) error {
+	if !pipelineRunFacts.IsRunning() {
```

Comment:
NIT: this check can be outside of this function, mainly do not call `gracefullyCancelPipelineRun` if `!pipelineRunFacts.IsRunning()` along with two other conditions in reconciler:

```
if cfg.FeatureFlags.EnableAPIFields == apisconfig.AlphaAPIFields && pr.IsGracefullyCancelled() {
```

The reason being, `pipelineRunFacts` can be avoided being in the list of params since its not used anywhere else besides this `if` condition in `gracefullyCancelPipelineRun`.

Reply:
Yes, that works well. Thanks!